core/src/main/java/org/elasticsearch/action/termvectors/TermVectorsResponse.java

```diff
@@ -104,6 +105,26 @@ public TermVectorsResponse(String index, String type, String id) {
     TermVectorsResponse() {
     }
 
+    public TermVectorsResponse(TermVectorsResult result) {
+        this.index = result.getConcreteIndex();
+        this.type = result.getType();
+        this.id = result.getId();
+
+        if (result.getDocVersion() != null) {
+            setDocVersion(result.getDocVersion());
+        }
+        setArtificial(result.getArtificial());
+        setExists(result.getExists());
+        try {
+            if (result.getTermVectorsByField() != null) {
+                setFields(result.getTermVectorsByField(), result.getSelectedFields(), result.getFlags(),
+                        result.getTopLevelFields(), result.getDfs(), result.getTermVectorsFilter());
+            }
+        } catch (Throwable ex) {
```

Comment:
what exception is this hiding? I think we shouldn't do this here and let just bubble up?